apps/meteor/app/livechat/server/lib/Contacts.ts

```diff
@@ -43,12 +55,14 @@ export const Contacts = {
 		updateUser.$set.phone = (phone && [{ phoneNumber: phone }]) || null;
 		updateUser.$set.visitorEmails = (visitorEmail && [{ address: visitorEmail }]) || null;
 
-		const allowedCF = LivechatCustomField.find({ scope: 'visitor' }, { fields: { _id: 1 } }).map(({ _id }) => _id);
+		const allowedCF = (LivechatCustomField.find({ scope: 'visitor' }, { fields: { _id: 1 } }) as ILivechatCustomField[]).map(
+			({ _id }) => _id,
+		);
 
 		const livechatData = Object.keys(customFields)
 			.filter((key) => allowedCF.includes(key) && customFields[key] !== '' && customFields[key] !== undefined)
 			.reduce((obj, key) => {
-				obj[key] = customFields[key];
+				(obj as any)[key] = customFields[key];
```

Comment:
I always wanted to try this: https://fettblog.eu/typescript-better-object-keys/ to have a typed Object.keys 👀 so we don't have to typecast as any. maybe you can give it a try 😬

Reply:
Interesting!! Thanks for sharing :hugs: Was able to apply this here :smile: